istioctl/cmd/analyze.go

```diff
@@ -250,63 +227,43 @@ istioctl analyze -L
 				fmt.Fprintln(cmd.ErrOrStderr())
 			}
 
-			// Filter outputMessages by specified level, and append a ref arg to the doc URL
-			var outputMessages diag.Messages
-			for _, m := range result.Messages {
-				if m.Type.Level().IsWorseThanOrEqualTo(outputLevel.Level) {
-					m.DocRef = "istioctl-analyze"
-					outputMessages = append(outputMessages, m)
-				}
+			// Append a ref arg to the doc URL, and filter outputMessages by specified level
+			outputMessages := result.Messages.SetDocRef("istioctl-analyze").FilterOutLowerThan(outputThreshold.Level)
```

Comment:
Now that the function names are there I think this is pretty understandable without the comment.